app/controllers/api/v1/search_results_controller.rb

```diff
@@ -8,26 +8,34 @@ class SearchResultsController < ApplicationController
 
       def create
         if upload_form.valid?
-          search_results = upload_form.save
+          search_result_list = upload_form.save
 
-          message = I18n.t('activemodel.notices.models.search_result.create')
-          search_results_serializer = SearchResultsSerializer.new(search_results, meta: { message: message })
+          upload_meta = upload_meta(search_result_list: search_result_list)
+          search_result_serializer = SearchResultListSerializer.new(search_result_list, upload_meta)
 
-          render(json: search_results_serializer, status: :created)
+          render(json: search_result_serializer, status: :created)
         else
           render_error(detail: upload_form.errors.full_messages, status: :unprocessable_entity)
         end
       end
 
       def index
         search_result_list = current_user.search_results.order(:id)
-        pagy, paginated_results = paginated_resources_for(search_result_list)
+        pagy, paginated_result_list = paginated_resources_for(search_result_list)
 
-        search_result_serializer = SearchResultsSerializer.new(paginated_results, meta: meta_from_pagy(pagy))
+        search_result_serializer = SearchResultListSerializer.new(paginated_result_list, meta: meta_from_pagy(pagy))
 
         render(json: search_result_serializer, status: :ok)
       end
 
+      def show
+        search_result = SearchResult.find(params[:id])
+
+        search_result_details_serializer = SearchResultSerializer.new(search_result)
```

Comment:
I think we can omit the `details` as `search_result` is enough 🤔 
```suggestion
        search_result_serializer = SearchResultSerializer.new(search_result)
```